src/main/java/org/opensearch/flowframework/common/WorkflowResources.java

```diff
@@ -57,7 +58,7 @@ public enum WorkflowResources {
     /** Workflow steps for creating an ingest-pipeline and associated created resource */
     CREATE_SEARCH_PIPELINE(CreateSearchPipelineStep.NAME, WorkflowResources.PIPELINE_ID, null), // TODO delete step
     /** Workflow steps for creating an index and associated created resource */
-    CREATE_INDEX(CreateIndexStep.NAME, WorkflowResources.INDEX_NAME, null), // TODO delete step
+    CREATE_INDEX(CreateIndexStep.NAME, WorkflowResources.INDEX_NAME, DeleteIndexStep.NAME),
```

Comment:
I'm really really really wary about hiding a data-removing delete index step inside an innocuously-named "deprovision workflow" API.

Is there a way we can require a parameter be passed here, kind of an "are you sure" check?

Reply:
+100 on this, I think having a API with just `deprovision` that could delete user data without being clear about it can  be bad. We already have this in some sense with deleting other resources but for index the data could be non-recoverable.

Reply:
I'm okay with no delete index at all, we just wont deprovision that, they have to do it manually

Reply:
Created https://github.com/opensearch-project/flow-framework/issues/579 to notify user to delete such resources manually and removed DeleteIndex step.